loki/transformations/parametrise.py

```diff
@@ -87,10 +87,60 @@
 from loki.ir import nodes as ir, Transformer, FindNodes
 from loki.tools.util import as_tuple, CaseInsensitiveDict
 
+from loki.transformations.utilities import single_variable_declaration
 from loki.transformations.inline import inline_constant_parameters
 
 
-__all__ = ['ParametriseTransformation']
+__all__ = ['ParametriseTransformation', 'parametrise_single_val_scalars']
+
+
+def parametrise_single_val_scalars(routine):
```

Comment:
The namings not ideal (as discussed offline). In particular, `parametrise` in the transformation refers to replacing values by constant parameters, whereas here we essentially mark local variables as compile-time constants. How about `declare_fixed_value_scalars_as_constants`?